src/sagas/projects.js

```diff
@@ -134,15 +138,31 @@ export function* projectExported({payload: {exportType}}) {
 }
 
 export function* saveCurrentProject() {
-  const userId = yield select(getCurrentUserId);
   const currentProject = yield select(getCurrentProject);
+  if (currentProject) {
+    yield* saveProjectWithKey(currentProject.projectKey);
+  }
+}
 
-  if (userId && currentProject && !isPristineProject(currentProject)) {
-    yield call(saveProject, userId, currentProject);
+function* saveProjectWithKey(projectKey) {
+  const state = yield select();
+  const userId = yield select(getCurrentUserId);
+  const project = yield call(getProject, state, {projectKey});
+  if (userId && !isPristineProject(project)) {
+    yield call(saveProject, userId, project);
     yield put(projectSuccessfullySaved());
   }
 }
 
+export function* archiveProject({payload: {projectKey}}) {
+  yield* saveProjectWithKey(projectKey);
+}
+
+export function* unArchiveProject({payload: {projectKey}}) {
+  const state = yield select();
+  yield call(saveProjectWithKey, state, projectKey);
```

Comment:
It doesn’t look like `saveProjectWithKey` takes `state` as a parameter?

Reply:
Also, the `yield` approach in `archiveProject` and the `call` approach here are both basically fine, but it seems like we should be consistent about it?